### ISSUE TITLE
rest-tool contact for discussion / next features / etc.

### ISSUE THREAD

**Issue by asmodehn** [tombenke/rest-tool#5]
Hello,

I am really interested in rest-tool, since I will need to implement a rest backend in the coming few months.

However I have many questions and I would like to be able to talk about it...
Is there a mailing list where we can discuss ? or how about setting up a https://gitter.im ?

Among my questions are :
- What are the differences with swagger ?
- I really like the code generation in node.js, with mock data to be able to quickly test a client on it. That's the main thing I am looking for, to be able to develop a client for prototyping without worrying too much about deployment issues at first.

However, how about embracing other projects, like RAML for the format of the configuration ? How about being able to generate a whole server project only from a simple config file ? How about supporting other languages ( python/tornado, erlang/leptus, etc. )

In short I think this is a very interesting project, and it should focus where other project don't, and play nice with them :) So I wanted to hear your thoughts / plans for it.

Anyway, many things to talk about...

**Comment by tombenke** [comment]
Hi,

First of all, thanks for using rest-tool, and thanks for suggesting gitter, that I have just opened for this project.

Short answers to your questions:
- The main difference with swagger is that I wanted to follow a language-neutral approach, in the meaning of focusing only on the REST interface and not on objects and classes, because that easily puts someone to the track of thinking in java, and in the OOP way, which accidentally might lead to tight coupling between the providers and consumers of the interface.
- The templates makes possible to generate clients not only in JavaScript, but theoretically in any programming languages. Just replace the original templates with yours, or write additional ones and refer them in the service.yml file, then the next test generator will use them.
- It is also possible to add server generator, using server-side templates, similar to the test generator.

Anyway, there is an other tool I made for creating project boilerplates, that is https://github.com/tombenke/kickoff. It is a more generic tool, which has similarities with the rest-tool. I may refactor the rest-tool to use the kickoff. We succesfully used these two tools in combination to generate full-stack application including Node.js backend and ExtJS frontend, with automatically generated siesta component tests.

**Comment by asmodehn** [comment]
Thanks for the quick answer:)
I am really interested by this multi-language approach. For me Javascript is a very simple and fast way to prototype a new software architecture, no matter the scale. But when you move to actual production, you might want to make other choices, for stability, speed, etc.
I think the key here is language/paradigm agnostic simple protocols, like REST.

As for javascript frameworks, I am used to dojo or jquery mostly, and dont have enough time to evaluate them all, but I ll have a look at kickoff sometime...

Well I guess I ll close this issue and I ll catch you on gitter :)
